actors/builtin/miner/miner_actor.go

```diff
@@ -188,44 +186,45 @@ func (a Actor) ChangeWorkerAddress(rt Runtime, params *ChangeWorkerAddressParams
 	}
 
 	var st State
-	isWorkerChange := false
 	rt.State().Transaction(&st, func() {
 		info := getMinerInfo(rt, &st)
 
 		// Only the Owner is allowed to change the newWorker and control addresses.
 		rt.ValidateImmediateCallerIs(info.Owner)
 
-		{
-			// save the new control addresses
-			info.ControlAddresses = controlAddrs
-		}
+		// save the new control addresses
+		info.ControlAddresses = controlAddrs
 
-		{
-			// save newWorker addr key change request
-			// This may replace another pending key change.
-			if newWorker != info.Worker {
-				isWorkerChange = true
-				effectiveEpoch = rt.CurrEpoch() + WorkerKeyChangeDelay
-
-				info.PendingWorkerKey = &WorkerKeyChange{
-					NewWorker:   newWorker,
-					EffectiveAt: effectiveEpoch,
-				}
+		// save newWorker addr key change request
+		// This may replace another pending key change.
+		if newWorker != info.Worker {
+			info.PendingWorkerKey = &WorkerKeyChange{
+				NewWorker:   newWorker,
+				EffectiveAt: rt.CurrEpoch() + WorkerKeyChangeDelay,
 			}
+		} else {
+			// Allow owner to cancel existing update by supplying existing worker address
+			info.PendingWorkerKey = nil
```

Comment:
good idea

Reply:
This has broken the prior behaviour of supporting "no change" to the worker key, while submitting an update to the control addresses. The old way didn't support cancelling a worker key change, but I'm not sure it should: cancelling a change (without delay) amounts to getting an instantaneous choice between two worker keys at the epoch before the change takes effect.

Please revert this for now so the rest of this PR isn't hung up on discussion about it. And to save the next person, add a comment explaining that specifying the worker key equal to the current key allows setting the control addreses without interfering with the worker key or any pending change to it.

Reply:
Ooh. This is right @anorth. Good catch.

Reply:
@sternhenri, @anorth the old behavior already has this instantaneous choice. If a miner had initiated a change to `addr2` 899 epochs ago, the miner could either allow the change to happen next epoch or make a change to `addr3` which would effectively cancel the change for another 900 epochs. 

If we want to avoid canceling, the condition should if a change exists rather than if the change is to the existing address. This means that an incorrect change will take a minimum of 1799 epochs to correct no matter if it's a revert or a change to a third address.

Reply:
mmh fair enough. Sorry hadn't paged in this exact thing. I expected we prevented any update to workerKey while a pending change was in progress...

Worth picking up in a new issue after merger I reckon

Reply:
I've pushed a change to prevent altering the pending worker key until the change has been effected. I can revert it. I've added an issue for future work at #1066.